### PR TITLE
Add Orientations for testing Hydrostatic balance

### PR DESCRIPTION
**Pull request by blallen**
### Description

Add Orientations to the ThreeDimensionalCompressableNavierStokesEquations so that one can determine the direction of gravity. Test that `FlatOrientation()` in a box gives the same result as `k̂ = [0, 0, 1]` for a stably stratified fluid. Also, refine some things about the buoyancy interpretation. (Buoyancy relation is for an ocean-like fluid at this point).

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by blallen** [comment]
bors try

**Comment by blallen** [comment]
bors try

File: test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    ρ₀(p, x, y, z) =
+        p.ρₒ * (1 - (p.α * p.g / p.cₛ^2) / 2 * (-p.θₒ * (1 - z / 4π))^2)
+    ρu₀(p, x...) = ρ₀(p, x...) * u₀(p, x...)
+    ρv₀(p, x...) = ρ₀(p, x...) * v₀(p, x...)
+    ρw₀(p, x...) = ρ₀(p, x...) * w₀(p, x...)
+    ρθ₀(p, x...) = ρ₀(p, x...) * θ₀(p, x...)
+
+    ρu⃗₀(p, x...) = @SVector [ρu₀(p, x...), ρv₀(p, x...), ρw₀(p, x...)]
+    initial_conditions = (ρ = ρ₀, ρu = ρu⃗₀, ρθ = ρθ₀)
```
Comment (charleskawczynski) [inline test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: More unrecognizable characters. Is this just me? cc @kpamnany 

Screen Shot 2021-03-16 at 2 40 53 PM

Comment (blallen) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: https://juliamono.netlify.app/ exists and it's not that much harder to install than julia itself

Comment (kpamnany) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: They are blank blocks for me as well. I don't think it's wise to require that someone install a font just to read our code.

Comment (bischtob) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Works fine for me :).

Comment (jkozdon) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Blank blocks for me.

Comment (blallen) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: https://github.com/CliMA/ClimateMachine.jl/blob/5165fc041efa68c02fd87f2fb0a3492ca2d05d06/src/Common/Orientations/Orientations.jl#L89
https://github.com/CliMA/ClimateMachine.jl/blob/5165fc041efa68c02fd87f2fb0a3492ca2d05d06/src/Atmos/Model/AtmosModel.jl#L461

I mean I can do this 
Screen Shot 2021-03-17 at 12 36 17
but I think that is a decrease in code readability

**Review by sandreza — APPROVED** [review]

File: test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl (hunk 1 above)
Comment (kpamnany) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Without the font, it's an even greater decrease in readability. See: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/AcceptableUnicode/

Comment (bischtob) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: I think the acceptable unicode page needs an update :). We need to be able to distinguish vectors from scalars.

Comment (charleskawczynski) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: I admit, it is helpful to have `\hat` sometimes.

Comment (bischtob) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: > Uh, okay. So let's just ignore them then.

Not, not ignore.

Comment (simonbyrne) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: @kpamnany @charleskawczynski what browser are you using? I see it fine in Firefox.

Comment (charleskawczynski) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Chrome

Comment (kpamnany) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Brave

Comment (simonbyrne) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Firefox apparently uses Menlo Regular for this item which renders it fine (also fine if I paste it into my terminal and use Menlo Regular). Can you check want font your browser is using? 

(I did it by right click on the text, click inspect element, then there was a "Fonts" tab which tells you what font it is)

Comment (charleskawczynski) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Interesting, Brave on my phone renders fine

Comment (simonbyrne) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Can you check what font it is using?

Comment (charleskawczynski) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: `font-family: SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace;`

Comment (kpamnany) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: Brave on desktop: `font-family: SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace;`. Not sure how to check it on the phone.

Comment (simonbyrne) [reply test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/test_buoyancy.jl:93]: That's the font list. Is there a way to check which font is actually used?

**Pull request merged**